lib/proj/get_proj.c

```diff
@@ -249,7 +249,8 @@ int pj_get_kv(struct pj_info *info, const struct Key_Value *in_proj_keys,
     alloc_options("type=crs");
 #endif
     pjc = proj_context_create();
-    if (!(pj = proj_create_argv(pjc, nopt, opt_in))) {
+    if ((info->srid && !(pj = proj_create(pjc, info->srid))) ||
+        (!info->srid && !(pj = proj_create_argv(pjc, nopt, opt_in)))) {
```

Comment:
I recommend to first create a PJ object and then recycle the code at https://github.com/OSGeo/grass/blob/main/lib/proj/do_proj.c#L332 to check if it is a `PJ_TYPE_BOUND_CRS`, in which case the source crs should be retrieved.

Applies not only to `pj_get_kv()`, but also to  `pj_get_string()`